cmd/collector/app/sanitizer/zipkin/span_sanitizer.go

```diff
@@ -121,3 +122,25 @@ func (s *parentIDSanitizer) Sanitize(span *zc.Span) *zc.Span {
 	span.ParentID = nil
 	return span
 }
+
+func NewErrorTagSanitizer() Sanitizer {
+	return &errorTagSanitizer{}
+}
+
+type errorTagSanitizer struct {
+	log spanLogger
+}
+
+func (s *errorTagSanitizer) Sanitize(span *zc.Span) *zc.Span {
+	for i := 0; i < len(span.BinaryAnnotations); i++ {
+		anno := span.BinaryAnnotations[i]
+		if strings.EqualFold("error", anno.Key) {
+			anno.AnnotationType = zc.AnnotationType_BOOL
+			anno.Value = []byte("error")
```

Comment:
I hadn't considered that. Older clients emitting zipkin directly might still be emitting Annotation{key: error, value: true} so we might have to add an extra check which only adds a log entry if the value is not "true" or True